src/Core/Usecase/Message/ReceiveMessage.php

```diff
@@ -111,118 +107,142 @@ public function setTargetedSurveyStateRepo(Entity\TargetedSurveyStateRepository
-			Kohana::$log->add(
-				Log::ERROR,
-				'Could not create new incoming message for contact_id: ' . print_r($contact_id, true)
-			);
-			throw new HTTP_Exception_400('Could not create new incoming message for contact_id: ' . $contact_id);
-		}
 		return $incomingMessageId;
 	}
 
 	/**
 	 * @param $contact_id
 	 * @param $survey_state_entity
 	 * @param $next_form_attribute
-	 * @return int|$incomingMessageId
-	 * @throws HTTP_Exception_400
+	 * @return int|$outgoingMessageId
 	 */
 	private function createOutgoingMessage($contact_id, $survey_state_entity, $next_form_attribute)
 	{
-		// create message that we will send to thhe user next
-		$newMessage = $this->repo->getEntity();
-		$messageState = array(
+		// Create new message to send next question to the user
+		$outgoingMessage = $this->repo->getEntity()->setState([
 			'contact_id' => $contact_id,
 			'post_id' => $survey_state_entity->post_id,
 			'title' => $next_form_attribute->label,
 			'message' => $next_form_attribute->label,
 			'status' => Message::PENDING,
-			'type' => 'sms',//FIXME
+			'type' => 'sms', // FIXME
 			'direction' => Message::OUTGOING
-		);
-		$newMessage->setState($messageState);
-		$this->outgoingMessageValidator->check($messageState);
-		$newMessageId = $this->repo->create($newMessage);
-		if (!$newMessageId) {
-			Kohana::$log->add(
-				Log::ERROR,
-				'Could not create new message for contact_id: ' . print_r($contact_id, true)
-			);
-			throw new HTTP_Exception_400('Could not create new outgoing message for contact_id: ' . $contact_id);
+		]);
+
+		// Verify its valid
+		// @todo not sure we even need to bother. If its not valid, all is lost.
+		if (!$this->outgoingMessageValidator->check($outgoingMessage->asArray())) {
+			$this->validatorError($outgoingMessage);
 		}
-		return $newMessageId;
+
+		// Save the message
+		$outgoingMessageId = $this->repo->create($outgoingMessage);
+
+		// But then continue anyway
+		return $outgoingMessageId;
 	}
 
 	/**
-	 * @param $contact_id
 	 * @param $incoming_message
-	 * @throws HTTP_Exception_400
 	 */
-	private function createTargetedSurveyMessages($contact_id, $incoming_message)
+	private function createTargetedSurveyMessages($incoming_message)
 	{
-		$surveyStateEntity = $this->targeted_survey_state_repo->getByContactId($contact_id);
-		$messageInSurveyState = clone $this->repo;
-		// ... attempt to load the entity
-		$messageInSurveyState = $messageInSurveyState->get($surveyStateEntity->message_id);
+		// Load the survey state for this contact
+		$surveyStateEntity = $this->targeted_survey_state_repo->getByContactId($incoming_message->contact_id);
+
+		// Attempt to load the previous message
+		$messageInSurveyState = $this->repo->get($surveyStateEntity->message_id);
+
+		// If we didn't find an outgoing message
+		// I'm not convinced this is right - if someone sends multiple incoming messages before the next outgoing one. But the survey isn't finished... shouldn't we add the messages to the original post?
```

Comment:
mmm. I think this might be weird if we add it to the contact's active post for the survey, since it'd be linked to the survey but not be a response to a question

(re-adding because I initially added the comment in a commit)